File: Chapter_12/app/src/main/java/com/rq/chapter12/MainActivity.java
package com.rq.chapter12;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ClassLoader loader = MainActivity.class.getClassLoader();
        while (loader != null) {
            Log.e("test", loader.toString());
            loader = loader.getParent();
        }
    }
}

File: Chapter_14/staticproxy/src/main/java/com/rq/staticproxy/MainActivity.kt
package com.rq.staticproxy

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val shop : IShop = Buyer()
        val buyerProxy = BuyerProxy(shop)
        buyerProxy.buy()
    }
}